c7n/policy.py

```diff
@@ -1074,6 +1074,15 @@ def evaluate(self, event=None):
             'policy': self.policy.data
         })
 
+        # Evaluating conditions from c7n-org file (e.g. account.tags) during Policy-Runtime
+        # causes the Policy skipping its execution as those account-specific conditions are
+        # only available during Policy-Deployment.
+        # See issue #7967.
+        # Solution is to remove those conditions during Runtime => "if self.env_vars == {}"
+        # As this dict is only populated via c7n-org during Deployment.
+        if not self.env_vars:
+            self.filters = [f for f in self.filters if not f.data['key'].startswith('account.')]
```

Comment:
rather than a "startswith" i think this should be "contains", given the fact that you can define a jmespath key with built-in functions like `ends_with(account.name, 'dev')` for instance

Reply:
@thisisshi But isn't account-level info passed from C7N-ORG a top-level key? Is there any scenario where `contains` would make sense? If there is, I would like to add it to the py-docs for this function and to my comment as well.

Reply:
it feels like we should instead propagate the values for evaluation at runtime, rather then attempting to remove the conditions. we can populate execution-options with the variables.

Reply:
It's too bad there isn't a clearer mechanism for deciding when a condition should be evaluated, either at deployment time or execution time.  `type: event` is only evaluated during execution time, which makes sense, given that's the only time the event exists.  I recently did some testing to help update the docs on how conditions work in a serverless policy and I'm not sure I can find a use-case where you'd want _anything_ other than `type: event` conditions being evaluated at execution time.  Even the case of the `now` key generally isn't needed as the event it's receiving should always have a `time` field showing when the event was generated.  Everything else you might filter on in a condition during the deployment would be identical during the execution phase, so I'm not sure I see the usefulness.  Should the logic for serverless be updated to:
* evaluate `type: value` conditions during deployment
* evaluate `type: event` conditions during execution
What use-case am I missing where this doesn't work?

Reply:
I've seen some examples for serverless for orgs with end of year change freezes, where they want to schedule and have the deployment but not be active till the communicated date passes.

@PratMis what's your experience been around conditions for server less?